packages/scripts/utils/config.js

```diff
@@ -309,35 +312,49 @@ function getWebpackEntryPoints( buildType ) {
 					entryPoints[ entryName ] = entryFilepath;
 				}
 			}
-
-			if ( Object.keys( entryPoints ).length > 0 ) {
-				return entryPoints;
-			}
 		}
 
 		// Don't do any further processing if this is a module build.
 		// This only respects *module block.json fields.
 		if ( buildType === 'module' ) {
-			return {};
+			return entryPoints;
 		}
 
-		// 3. Checks whether a standard file name can be detected in the defined source directory,
-		//  and converts the discovered file to entry point.
-		const [ entryFile ] = glob( 'index.[jt]s?(x)', {
-			absolute: true,
-			cwd: fromProjectRoot( getWordPressSrcDirectory() ),
-		} );
+		const hasBlockJsonInRoot = blockMetadataFiles.includes(
+			join(
+				fromProjectRoot( getWordPressSrcDirectory() ),
+				'block.json'
+			).replace( /\\/g, '/' )
+		);
+		if ( ! hasBlockJsonInRoot ) {
+			// 3. Checks whether a standard JavaScript file name can be detected in the defined source directory,
+			//    and converts the discovered file to an entry point.
+			const [ entryJavaScriptFile ] = glob( 'index.?(m)[jt]s?(x)', {
+				absolute: true,
+				cwd: fromProjectRoot( getWordPressSrcDirectory() ),
```

Comment:
I'd support the `c` (commonJS) prefix here, but I'd probably remove and reserve `m` (module) prefix as a script-modules indicator. What do you think?

Related: #65271

Reply:
Sounds good to me 👍